### PR TITLE
Feature: Website Refactor

### LINKED ISSUE #77: site refactor
optimize all pages for mobile
modularize where possible
general code cleanup

### PR DESCRIPTION
**Pull request by shredtechular**
This PR fixes #77 and accomplishes a number of different things:

# Items of particular importance
- While doing the refactor, I thought it would be good to also make the website's page URLs more concise and semantic for user experience and SEO gains.  If that causes any issues by external linking we may want to discuss reverting this semantic URL renaming or figure out a redirect solution.  I think it is important to point out though.
-  This refactor does include the implementation of the Netlify CMS.  I will plan to demo this to @j00bar and @howtohumanbetter to finalize.  If the team determines this helpful, I'll write up some quick "how-to" documentation for editing content.  If we want to explore other CMS ideas, that is fine too and can remove this piece.  Since the functionality is broken up into modules it should port much easier to other CMS platforms if desired.

# Accomplishments
## Code Cleanup
The refactor has accomplished the following for code cleanup:
-  Removal of unused CSS, Images, and some JS (still could use a little more untanglement, but this should suffice for moving forward). 
- A more formal directory structure and description in a README file
- Finally, refactored some of the HTML to remove redundant and unnecessary code and keeping things as semantic as possible

## Mobile Optimzation
The refactor has accomplished the following for mobile optimization:
- All of the code, especially the CSS, is now designed based on a mobile-first approach.  There are numerous advantages of this approach, including ease of maintainability for a fully responsive website, 
but the site should now perform and look much better across the board now.
- Each page was reviewed and tweaked to look great on mobile.  Included in these updates are responsive fixes for the footer and a much needed revamp to make all content align within consistent and slimmed down "gutters".

## Modularity
The refactor is now much more modular as well.  Instead of having huge HTML files and Jekyll basically gluing on a header and footer.  The code actually uses templating logic to dynamically build the pages.  This is necessary for maintainability, and CMS purposes, so this is a very key accomplishment.

## Other Optimizations
**The refactor has resulted in ~36% reduction in total page size across the site and ~21% decrease in total page load time... Please see below for specific page optimizations based on Googe PageSpeed and WebPageTest:
<kbd>
screen shot 2018-03-04 at 9 27 52 pm
</kbd>

## Future Additional Optimizations
- May want to revisit how the mobile menu works and make it more "sticky".  Kept the menu as is for refactor purposes.  Re-vamping the mobile menu is being considered outside the scope of the refactor.
- The JS files are all still separate files so separate requests and unminified.  Depending where the site is being hosted, there could be some additional performance benefits for doing this.  Unfortunately, native Jekyll doesn't handle this out of the box, so we would need a plugin to accomplish this or a NPM/Grunt/Gulp task, but that would need to be executed before a build.  So, this should be further discussed/planned before executing as it would have impacts to the CI process.

### CODE REVIEW

File: _includes/sections/events.html [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    <div class="container clearfix">
+      <div class="row display-flex">
+
+        {% for event in events %}
+        
+          <div class="col-md-12 bottommargin">
+            <div class="clearfix">
+              <div class="event-image">
+                <img src="{{ site.baseurl }}{{ event.image }}" alt="Event">
+                <div class="event-date">06<span>May</span></div>
```
Comment (j00bar) [inline _includes/sections/events.html]: Hard coded content.

**Comment by j00bar** [comment]
Note to @j00bar: Check for CloudFlare redirect support for URL changes

File: _includes/sections/parallax.html [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  
+--> 
+{% endcomment %}
+
+{% assign laxes = site.sections |where:"type","parallax" |where:"title",include.title %}
+
+{% for lax in laxes %}
+
+
+<div id="weCareBecause" class="section parallax nobottommargin" data-stellar-background-ratio="0.4" style="background-image: url({{ lax.bg-img }});">
```
Comment (j00bar) [inline _includes/sections/parallax.html]: Can't have an `id` in a for loop.

File: _includes/sections/hero.html [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  picture and text/video in a 2 column layout.  Currently on the Home page.
+  
+--> 
+{% endcomment %}
+
+{% assign heros = site.sections|where:"type","hero"|where:"title",include.title %}
+
+{% for hero in heros %}
+
+<section id="slider" class="slider-parallax full-screen dark">
```
Comment (j00bar) [inline _includes/sections/hero.html]: Can't have an `id` in a for loop.

File: _includes/sections/events.html (hunk 1 above)
Comment (shredtechular) [reply _includes/sections/events.html]: This is fixed now in 4f80ac0dc1dff3c90815d53ea14d7116661745ff

File: _includes/sections/hero.html (hunk 3 above)
Comment (shredtechular) [reply _includes/sections/hero.html]: As we discussed in our review meeting, and after trying to remove the id, this did break the js incorporated for the hero.  I don't think it is likely to have 2 heros on one page so I don't think this should be a showstopper.  I think we could add a future improvement task for cleaning up the js more though.  I did what I could with it for purposes of the refactor, but there is still more to untangle when it comes to the js due to the theme.

File: _includes/sections/parallax.html (hunk 2 above)
Comment (shredtechular) [reply _includes/sections/parallax.html]: Good catch!  This is fixed in bf288da1453f

**Pull request merged**